django/contrib/auth/hashers.py

```diff
@@ -201,6 +202,15 @@ def encode(self, password, salt):
         """
         raise NotImplementedError('subclasses of BasePasswordHasher must provide an encode() method')
 
+    def decode(self, encoded):
+        """
+        Return a decoded version of the database value.
+
+        The result is a dictionary of string value pairs used for checking
+        salt size, iteration counts, and so on.
+        """
+        return dict()
```

Comment:
Should always prefer a literal.

```suggestion
        return {}
```

Reply:
Ok, updated.